pyproj/_crs.pyx

```diff
@@ -387,15 +517,24 @@ cdef class _CRS:
         """
         Returns
         -------
-        pyproj.CRS: The datum as a CRS.
+        Datum: The datum.
         """
-        cdef PJ *projobj = proj_crs_get_datum(self.projctx, self.projobj)
-        if projobj == NULL:
-            return None
-        try:
-            return self.__class__(_to_wkt(self.projctx, projobj))
-        finally:
-            proj_destroy(projobj) # deallocate temp proj
+        if self._datum is not None:
+            return self._datum
+        self._datum = Datum.create(self.projctx, self.projobj, horizontal=False)
+        return self._datum
+
+    @property
+    def horizontal_datum(self):
```

Comment:
I am not very familiar with "horizontal datum" (and also don't find much about that in the ISO specs or proj docs), but when is datum and horizontal datum different?

Eg for `pyproj.CRS.from_epsg(3857)` (or some others I tried) they give me the same.

(just wondering if it is needed, at least to start with, to have both attributes)

Reply:
Honestly, I am not familiar either. From the code, it seemed datum only works with a single CRS whereas horizontal datum works for a compound CRS. Maybe the proj tests have an example.

Reply:
Removed for now